src/transforms/kubernetes/mod.rs

```diff
@@ -1 +1,552 @@
 pub mod watch_client;
+
+use self::watch_client::{ClientConfig, PodEvent, RuntimeError, Version, WatchClient};
+use super::Transform;
+use crate::{
+    event::{self, Event, Value},
+    sources::kubernetes::POD_UID,
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+};
+use bytes::Bytes;
+use evmap::{ReadHandle, WriteHandle};
+use futures::stream::Stream;
+use futures03::compat::Future01CompatExt;
+use k8s_openapi::api::core::v1::Pod;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::collections::VecDeque;
+use std::time::{Duration, Instant};
+use string_cache::DefaultAtom as Atom;
+use tokio::timer::Delay;
+
+// *********************** Defined by Vector **************************** //
+/// Node name `spec.nodeName` of Vector pod passed down with Downward API.
+const NODE_NAME_ENV: &str = "VECTOR_NODE_NAME";
+
+/// If watcher errors, for how long will we wait before trying again.
+const RETRY_TIMEOUT: Duration = Duration::from_secs(2);
+
+/// For how long will we hold on to pod's metadata after the pod has been deleted.
+///
+/// 10min sounds a lot, but a metadata entry averages around 300B and in an extreme
+/// scenario when a pod is deleted every second, we would have ~200kB of probably
+/// not usefull data. Which is acceptable. The benefit of such a long delay
+/// is that we will certainly* have the metadata while we are processing the
+/// remaining logs from the deleted pod.
+const DELETE_DELAY: Duration = Duration::from_secs(10 * 60);
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct KubePodMetadata {
```

Comment:
`retry_timeout` would, but `DELETE_DELAY` won't as it doesn't have an influence on the issue because it's only used after the api server is done with changing Pod metadata. 

And do you think that on large clusters deploying Vector with this transform could be an issue? 
[Related](https://github.com/timberio/vector/pull/1888#issuecomment-591079624)

Reply:
the fluentd plugin has a cache_ttl option so it may just make sense to provide the ability to configure it.

Reply:
True